### PR TITLE
[ON HOLD] Shadowlings 2.0

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
experiment: SHADEOWLEINGZ!
/:cl:

As of now, it 100% works as far as I can test

_**POLL ABOUT GLARE**_: http://www.strawpoll.me/13584514

Video showing off new shadow walk: https://streamable.com/lfs7x

Balance changes made:

* thrall's lesser glare now only makes the target confused
* removed life drain
* required thralls for accession scales with player count
* shadow walk jaunt ability was replaced with the passive ability to move fast in darkness, go through walls if it is dark on the other side, but move very slowly in the light as you take damage
* enthralling implanted people takes _**65**_ seconds more than non-implanted people
* you can kill idiotic ascendant shadowlings with the singularity

also, slings and thralls now have their own antag sound, like traitors/nukeops/cult/etc

### CODE REVIEW

**Comment by ChangelingRain** [comment]
hey do you remember the freeze

**Comment by MoreRobustThanYou** [comment]
> I just want this open for discussion and review until the freeze passes

File: code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 1]
```diff
@@ -0,0 +1,899 @@
...
+	for(var/obj/item/F in H)
+		blacklistLuminosity += extinguishItem(F)
+	H.set_light(blacklistLuminosity) //I hate lightcode for making me do it this way
+
+/obj/effect/proc_holder/spell/aoe_turf/veil/cast(list/targets,mob/user = usr)
+	if(!shadowling_check(user) && !admin_override)
+		revert_cast()
+		return
+	user << "<span class='shadowling'>You silently disable all nearby lights.</span>"
+	for(var/turf/T in targets)
```
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: this must apply proper procs to all possible lit objects, rather than set_light(0).

**Comment by qbmax32** [comment]
why does the policy discussion forum exist

File: code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 1 above)
Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: And what proc would that be?

**Comment by MoreRobustThanYou** [comment]
@qbmax32 lel i don't use the forums

File: code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: I don't know, there's a lot of objects that produce light. Most don't have a proc for it, and you'll need to make one.

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: before it uses SetLuminosity, also this seems to work, I tested it, the current method works

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+/proc/is_shadow(var/mob/living/M)
+	return istype(M) && M.mind && SSticker && SSticker.mode && (M.mind in SSticker.mode.shadows)
+
+
+/datum/game_mode/shadowling
+	name = "shadowling"
+	config_tag = "shadowling"
+	antag_flag = ROLE_SHADOWLING
+	//required_players = 30
```
Comment (McBawbaggings) [inline code/game/gamemodes/shadowling/shadowling.dm]: i assume this to be uncommented yeah?

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling.dm]: It has been, but due to this being closed, the latest commits don't show on the PR

File: code/game/gamemodes/shadowling/shadowling.dm [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+		thralls += new_thrall_mind
+		new_thrall_mind.special_role = "thrall"
+		message_admins("[key_name_admin(new_thrall_mind.current)] was enthralled by a shadowling!")
+		log_game("[key_name(new_thrall_mind.current)] was enthralled by a shadowling!")
+
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/lesser_shadowling_hivemind(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/lesser_glare(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/lesser_shadow_walk(null))
+		new_thrall_mind.AddSpell(new /obj/effect/proc_holder/spell/self/thrall_vision(null))
+		new_thrall_mind.current << "<span class='shadowling'><b>You see the truth. Reality has been torn away and you realize what a fool you've been.</b></span>"
```
Comment (McBawbaggings) [inline code/game/gamemodes/shadowling/shadowling.dm]: to_chat() is to be used

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling.dm]: Stop looking at the PR's page, look directly at the branch. This has been fixed!!

Comment (MoreRobustThanYou) [inline code/game/gamemodes/shadowling/shadowling.dm]: But yeah, all the <<'s, except for sound, which still works, has been refactored to use to_chat

**Comment by MoreRobustThanYou** [comment]
But from the branch, it 100% works now, as far as I can test without another player.

**Comment by MoreRobustThanYou** [comment]
@ChangelingRain would it be possible to open this for a split second to allow for the PR page to show the latest changes so it'd be easier to review?

**Pull request closed without merging**